### ISSUE TITLE
Extra class added with django-autocomplete-light select that breaks it.

### ISSUE THREAD

**Issue by glennbach** [django-crispy-forms/django-crispy-forms#1067]
* Package version: 1.9.2
* Django version: 3.0.8
* Python version: 3.7.9
* Template pack: Bootstrap4

### Description:

I'm using this with `django-autocomplete-light`. If I just display the plain form with:

```
{{add_member_form}}
```

everything works perfectly. If, I use crispy forms:

```
{% crispy add_member_form %}
```

then an extra class `select2` is added to the select, which breaks it. Where is that class being added? Can I remove it? Any suggestions? Thanks.

### Code:

My form:

```
class AddMemberForm(forms.Form):

    member = UserAutocompleteChoiceField(
        choices=[],
        widget=autocomplete.Select2(
            url='core:user--autocomplete',
            attrs={
                'autofocus': 'autofocus',
                'data-placeholder': 'Enter a name search here ...',
                'data-minimum-input-length': 3,
                'dropdownParent': "#addMemberModal"
            }
        )
    )

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.helper = FormHelper()
        self.helper.form_id = 'id-addmemberform'
        self.helper.form_class = 'form'
        self.helper.form_method = 'post'
        self.helper.form_action = reverse('core:group--create')
        self.helper.include_media = False
        self.helper.add_input(Submit('submit', 'Add'))

```

Actual HTML generated:

```
<form action="..." class="form" id="id-addmemberform" method="post" name="id-addmemberform">
	<input type="hidden" name="csrfmiddlewaretoken" value="...">
	<div id="div_id_member" class="form-group">
		<label for="id_member" class="requiredField">Member<span class="asteriskField">*</span></label>
		<div class="">
			<select name="member" autofocus="autofocus" data-placeholder="Enter a name search here ..." data-minimum-input-length="3" dropdownparent="#addMemberModal" class="select2 form-control" id="id_member" data-autocomplete-light-url="..." data-autocomplete-light-function="select2" data-autocomplete-light-language="en">
			</select>
		</div>
	</div>
	<div class="form-group">
		<div class="">
			<input type="submit" name="submit" value="Add" class="btn btn-primary" id="submit-id-submit">
		</div>
	</div>
</form>

```

**Comment by smithdc1** [comment]
The Select2 class is not being added by crispy forms. Is there some other js on the page that is adding this?

**Comment by glennbach** [comment]
I'm just confused then because when i just display the field with:

```
{{add_member_form}}
```

it doesn't have the class. I would think that some other js would do the same in this case.

**Comment by smithdc1** [comment]
Yes, I'm not sure.

See here, another example where js elsewhere was impacting the rendering of
a form, in that case it was changing a button.


https://github.com/django-crispy-forms/django-crispy-forms/issues/1016


A more relevant an example of select2 and bootstrap not working together
well?

https://stackoverflow.com/questions/59988470/django-select2widget-not-getting-styled-properly-with-crispy-forms

If these examples don't help. I'd try asking on one of the other support
channels as you may find someone who has used these tools together.

- Django forum. https://forum.djangoproject.com/ ( ken is a hero)
- stack overflow
- Django users mailing list. https://groups.google.com/g/django-users

**Comment by glennbach** [comment]
Thanks for heading me in the right direction.

**Comment by smithdc1** [comment]
No problem. I wish you all the best with your project. 😀

**Comment by glennbach** [comment]
One quick followup. I tried printing the html in my view with:

```
print(render_crispy_form(context['add_member_form']))
```

and it shows:

```
class="select2 form-control"
```

doesn't this indicate that the `select2` class is not being added with JS?

Thanks.

**Comment by smithdc1** [comment]
Yes. I guess Dal is adding it somewhere. There was an issue at gal that sounds similar to your issue? 🤔

https://github.com/yourlabs/django-autocomplete-light/issues/698

**Comment by glennbach** [comment]
I figured it out, finally. In CrispyFieldNode from templatetags.crispy_forms_field, you set the class_name to:

```
class_name = widget.__class__.__name__.lower()
```

So, since the widget is Select2, the class is set to select2, which apparently screws up the javascript. To fix it, I just had to add the following to my settings:

```
CRISPY_CLASS_CONVERTERS = {'select2':''}
```

so it doesn't do that. I'm mentioning this just in case someone else might have this issue.
